common-money.lic

```diff
@@ -69,6 +106,56 @@ module DRCM
     check_wealth(hometown_currency(hometown))
   end
 
+  def get_total_wealth
+    # This method captures your current total on-hand wealth
+    # and returns a hash representing the numerical value in
+    # coppers of each currency.    
+
+    # Set up variables to capture the value in coppers of each currency
+    # Set to zero so that, if we have, for example, "No Lirums"
+    # we simply return the initialized value of 0.
+    kronars = 0
+    lirums = 0
+    dokoras = 0
+
+    # Grab the character's wealth, pausing a bit
+    # then grabbing a sufficient number of lines
+    # to ensure we get all the output taking into
+    # account other random scroll text.
+    # Reversing the lines ensures we are processing
+    # the most recent output from 'wealth', in case
+    # reget were to grab output from back-to-back calls.
+    DRC.bput("wealth", "Wealth")
+    pause 0.5
+    wealth_lines = reget(10).map(&:strip).reverse
+
+    # We've reversed the reget array. Now we'll iterate over it and capture
+    # each line after we recognize we've hit the Wealth block.
+    wealth_lines.each do |line|
+      case line
+      when /^Wealth:/i
+        # This is the start of our Wealth lines.
+        # We don't need to parse this line. Break out of loop.
+        break
+      when /\(\d+ copper Kronars\)/i
+        kronars = line.scan(/\d+/).first.to_i
+      when /\(\d+ copper Lirums\)/i
+        lirums = line.scan(/\d+/).first.to_i
+      when /\(\d+ copper Dokoras\)/i
+        dokoras = line.scan(/\d+/).first.to_i
+      end
```

Comment:
Apologies for the bug my suggestion introduced. I forgot that at this point the context of `line` is the entire line, not just the regex pattern that was matched. In the following, the use of `Regexp.last_match` ensures we're looking at only the part of the line that indeed matched the pattern. I also named the capturing group that matched the coins so that you can grab them right out of the matched result, no further scanning or parsing necessary.

My character is out hunting at moment so I haven't tested it, but I'm pretty confident with the below snippet.

```ruby
      when /\((?<amount>\d+) copper Kronars\)/i
        kronars = Regexp.last_match[:amount].to_i
      when /\((?<amount>\d+) copper Lirums\)/i
        lirums = Regexp.last_match[:amount].to_i
      when /\((?<amount>\d+) copper Dokoras\)/i
        dokoras = Regexp.last_match[:amount].to_i
      end
```

This next bit isn't necessary, just sharing as FYI. You could get fancier with naming capturing groups to get the currency, too, and then be more dynamic with populating the total wealth map result.

```ruby
      when /\((?<amount>\d+) copper (?<currency>Kronars|Lirums|Dokoras)\)/i
        total_wealth[Regexp.last_match[:currency].downcase] = Regexp.last_match[:amount].to_i
      end

```

Reply:
You don't have to use what I proposed above, just trying to atone for my mistake :)

Reply:
Haven't quite wrapped my head around these fancier versions. But I did commit a fix that is within my current ability.